### DISCUSSION TITLE
Terms of Service Consent Pop-up for new users before access to application

### DISCUSSION THREAD

**Discussion by javi-sc** [zitadel/zitadel#6747] [Q&A]
We are using self-hosted Zitadel for our SaaS application, and I am trying to sort out how to "enforce" users accepting Terms of Service. 
In my eyes, it should work something like: 
1. User signs in for the first time. 
2. User Accepts Terms of Service
3. User gets access to the application. 
Based on the documentation (see image) seems possible, but I do not find the way to configure it. 

![image](https://github.com/zitadel/zitadel/assets/121081259/e008b3fb-c541-4c28-9894-08e791875af5)
I have found this discussion that seems relevant: https://github.com/zitadel/zitadel/discussions/4383?sort=new

**Comment by hifabienne** [comment]
Hi @javi-sc 
The question for accepting the terms and services is currently only shown when the user registers.

**Reply by javi-sc** [reply]
Hi @hifabienne , 
We are using an external Identity provider (AzureAD) for some of the users, and when I entered with the SSO experience, the pop-up of terms of service did not appear. Is there a way to enable it also for this situation, or should it also work?
I will test with newly registered users to see if that works. 
Thanks

**Reply by hifabienne** [reply]
when you register a new user with an external idp you also get the accept tos and privacy policy check box
